File: patterns/Python/pattern17.py
"""
Generate the following pattern
4444444
4333334
4322234
4321234
4322234
4333334
4444444
"""

def pattern16(n):
    for i in range(2*n-1):
        for j in range(2*n-1):
            top = i
            left = j
            right = 2*(n-1)-j
            bottom = 2*(n-1)-i
            
            ans = min(top, bottom, left, right)
            print(n-ans, end="")
                   
        print("")
   
#Driver Code 
def main():
    n = int(input("Enter the number of rows: "))
    print("Here is the generated pattern\n")
    pattern16(n)
    
if __name__ == "__main__":
    main()
           

File: patterns/C++/pattern17.cpp
/*
4444444
4333334
4322234
4321234
4322234
4333334
4444444
*/

#include <iostream>

using namespace std;

void pattern16(int n) {
    for (int i = 0; i < 2 * n - 1; i++) {
        for (int j = 0; j < 2 * n - 1; j++) {
            int top = i;
            int left = j;
            int right = 2 * (n - 1) - j;
            int bottom = 2 * (n - 1) - i;

            int ans = min(min(top, bottom), min(left, right));
            cout << n - ans;
        }
        cout << endl;
    }
}

int main() {
    int n;
    cout << "Enter the value of n: ";
    cin >> n;

    cout << "Here is the generated pattern\n";
    pattern16(n);

    return 0;
}